### ISSUE TITLE
feat(config): support intellisense in .syncpackrc.js

### ISSUE THREAD

**Issue by jamiehaywood** [JamieMason/syncpack#130]
## Description

In lieu of having a TS config file, it would be useful to add a types key to the `package.json` so that we can consume the config types i.e:

```js
/**
 * @typedef {import('syncpack').Syncpack.Config.Public} SyncpackConfig
 * @type {SyncpackConfig}
 **/
export default {
  customTypes: [],
  dependencyTypes: ['dev', 'peer', 'prod', 'workspace'],
  filter: '.',
...
```

## Suggested Solution
Should be as simple as adding a `types` key to the `package.json` pointing to `dist/types.d.ts`.

## Help Needed
N/A

**Comment by JamieMason** [comment]
Thanks Jamie, agree and been meaning to do this.

**Comment by jamiehaywood** [comment]
a follow up to this (can raise a separate ticket) is potentially adding a JSON schema to the `dist` so it can be referenced in the SyncPack config 🙂

**Comment by JamieMason** [comment]
Got this working locally, it'll look like this:

```js
// @ts-check

/** @type {import("syncpack").RcFile} */
const config = {
  versionGroups: [
    {
      dependencies: ["@types/node"],
      packages: ["**"],
      pinVersion: "14.18.36",
    },
    {
      dependencies: ["chalk"],
      packages: ["**"],
      pinVersion: "4.1.2",
    },
  ],
};

module.exports = config;
```

This should hopefully help fix https://github.com/JamieMason/syncpack/issues/132 too, as the types complain if you mix properties from different version groups for example.

**Comment by JamieMason** [comment]
Released in [syncpack@10.0.0](https://github.com/JamieMason/syncpack/releases/tag/10.0.0), please let me know if you run into any issues.
